modules/mobile/app/controllers/mobile/v0/demographics_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'va_profile/demographics/service'
+
+module Mobile
+  module V0
+    class DemographicsController < ApplicationController
+      before_action { authorize :demographics, :access_update? }
+      before_action { authorize :mpi, :queryable? }
+
+      def index
+        response = service.get_demographics
+        raise_error(response) if response.status != 200
+        render json: Mobile::V0::DemographicsSerializer.new(@current_user.icn, response)
+      end
+
+      private
+
+      def raise_error(response)
+        case response.status
+        when 400
+          raise Common::Exceptions::BadRequest
+        when 404
+          raise Common::Exceptions::RecordNotFound, @current_user.icn
+        end
+
+        raise Common::Exceptions::BackendServiceException
```

Comment:
the VA profile Demographics service will use partial data using user fields if the demographics service fails. Unfortunately, the fields we want are only available through the upstream service so partial data is as good as no data in our case. this coverts the partial data into an error. There may be more errors but these are the ones we have test data for and probably the ones most likely to come up.